<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>10</y>
      <width>661</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Man-in-the-middle Attack</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Alice_image">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>100</y>
      <width>141</width>
      <height>141</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/coding (1).png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Bob_image">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>100</y>
      <width>141</width>
      <height>141</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/coding.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="message_image">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>190</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/envelope.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Alice_name">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>80</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Alice</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Bob_name">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>80</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Bob</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>250</y>
      <width>160</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="AliceProperties">
     <item row="0" column="0">
      <widget class="QLabel" name="PKey_label">
       <property name="text">
        <string>Private key:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="A_PKey_input"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>250</y>
      <width>160</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="BobProperties">
     <item row="0" column="1">
      <widget class="QLineEdit" name="B_PKey_input"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="B_PKey_label">
       <property name="text">
        <string>Private key:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>320</y>
      <width>201</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Public keys</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>180</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="1">
       <widget class="QLineEdit" name="G_input"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="p_label">
        <property name="text">
         <string>Prime numer p:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="G_label">
        <property name="text">
         <string>Generator number g:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="p_input"/>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QLabel" name="Eve_label">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>100</y>
      <width>131</width>
      <height>141</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/hacker.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Eve_name">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>80</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Eve</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>430</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Start exchange key</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="E_A_PKey_input">
    <property name="geometry">
     <rect>
      <x>297</x>
      <y>90</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="E_A_PKey_label">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>90</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Alice - Eve private key:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="E_B_PKey_input">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>90</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="E_B_PKey_label">
    <property name="geometry">
     <rect>
      <x>455</x>
      <y>90</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Bob - Eve private key:</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>310</y>
      <width>201</width>
      <height>231</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="logLayout">
     <item>
      <widget class="QLabel" name="logLabel">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Log</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="Log"/>
     </item>
    </layout>
   </widget>
   <zorder>message_image</zorder>
   <zorder>label</zorder>
   <zorder>Alice_image</zorder>
   <zorder>Bob_image</zorder>
   <zorder>Alice_name</zorder>
   <zorder>Bob_name</zorder>
   <zorder>gridLayoutWidget_2</zorder>
   <zorder>gridLayoutWidget_3</zorder>
   <zorder>groupBox</zorder>
   <zorder>Eve_label</zorder>
   <zorder>Eve_name</zorder>
   <zorder>pushButton</zorder>
   <zorder>E_A_PKey_input</zorder>
   <zorder>E_A_PKey_label</zorder>
   <zorder>E_B_PKey_input</zorder>
   <zorder>E_B_PKey_label</zorder>
   <zorder>verticalLayoutWidget</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
